lib/dao.js

```diff
@@ -466,6 +466,10 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
   assert(typeof options === 'object', 'The options argument must be an object');
   assert(typeof cb === 'function', 'The cb argument must be a function');
 
+  if (Array.isArray(data)) {
+    return cb(new Error('updateOrCreate does not support bulk mode or any array input'));
```

Comment:
I think you need to return `cb.promise` in order to keep supporting Promise API.